public/js/libs/helper.js

```diff
@@ -0,0 +1,129 @@
+const plotSingle = (graph, layer, transition = true) => {
+    graph.class_keep = graph.clickedLegend;
+    graph.idx = graph.legendIdentityArray.indexOf(graph.class_keep);
+
+    // Erase all but selected bars by setting opacity to 0
+    for (let i = 0; i < graph.legendIdentityArray.length; i++) {
+        if (graph.legendIdentityArray[i] != graph.class_keep) {
+            if (!transition) {
+                d3.selectAll(`.${graph.legendIdentityArray[i]}`).style("opacity", 0);
+            } else {
+                d3.selectAll(`.${graph.legendIdentityArray[i]}`).transition().duration(500).style("opacity", 0);
+            }
+        }
+    }
+
+    // Lower the bars to start on x-axis
+    graph.y_orig = []; // to store original y-posn
+    layer.selectAll("rect")._groups[graph.idx].forEach(function (d, i, n) {
+        // Get height and y posn of base bar and selected bar
+        let h_keep = d3.select(d).attr("height");
+        let y_keep = d3.select(d).attr("y");
+        // Store y_base in array to restore plot
+        graph.y_orig.push(y_keep);
+
+        let h_base, y_base;
+        layer.selectAll("rect")._groups[0].forEach(function (d, i, n) {
```

Comment:
Is there a way of doing this without poking around at the graph's private internal state (_groups)?